torch/_dynamo/eval_frame.py

```diff
@@ -98,6 +98,11 @@ def __setattr__(self, name, value):
         super().__setattr__(name, value)
 
     def __call__(self, *args, **kwargs):
+        if hasattr(self._orig_mod, "_initialize_hook"):
+            # In the case of a lazy module, we want to run
+            # the pre-hooks which initialize it
+            assert len(kwargs) == 0
+            self._orig_mod._infer_parameters(self._orig_mod, args)
```

Comment:
wait, i think this does the opposite of what I originally pointed out.  In eager, does a lazy module allow itself to be 're-infered' after the first time?  or is it a 1-shot deal?

Reply:
It's a 1-shot deal, but I think my implementation is consistent with eager mode. Look at the difference between here and the other places where I call ```_infer_parameters```. I checked ```hasattr(nnmodule, "_initialize_hook")``` to avoid "re-infered" in other places, but here I didn't because this is the entry point of calling ```torch.compile```, I think we disallow to call ```torch.compile``` on an already compiled function.